### ISSUE TITLE
Add scroll physics configuration to `ScrollBar`

### ISSUE THREAD

**Issue by angelosilvestre** [flutter/flutter#139666]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Use case

On the super_editor package, we use a `SingleChildScrollView` to display the scrollable content  However, we handle the scrolling gestures ourselves, so we prevent the `SingleChildScrollView` from handling the gestures by configuring it with a `NeverScrollableScrollPhysics`, like the following:

```dart
  SingleChildScrollView(
    controller: widget.scrollController,
    physics: const NeverScrollableScrollPhysics(),
    child: ...
   )
```

However, this prevents the users from interacting with the scrollbar, even if we wrap the `SingleChildScrollView` with a `Scrollbar`.

```dart 
  Scrollbar(
    controller: widget.scrollController,
    child: ScrollConfiguration(
      behavior: ScrollConfiguration.of(context).copyWith(scrollbars: false),
      child: SingleChildScrollView(
        controller: widget.scrollController,
        physics: const NeverScrollableScrollPhysics(),
        child: ...
        ),
     ),
   )
```

This happens because the scrollbar checks if user interaction is accepted using the `ScrollPosition`'s scroll physics, which, in this case, doesn't accept user interaction.

### Proposal

Add an optional `physics` property to the `Scrollbar` so we can use a different `ScrollPhysics` that allows user interaction.

**Comment by Piinks** [comment]
I don't think this is something we are likely to support. I do not know if ScrollPhysics on a Scrollbar will make semantic sense, what happens when the physics of the position and the physics of the scrollbar conflict? Will users expect physics set on a scrollbar to be reflected in the scroll position? The scrollbar does not implement the physics, it just queries whether or not scrolling is allowed, which in this case, it is not allowed. NeverScrollableScrollPhysics is pretty clear that is does not allow scrolling. I don't know that we should complicate that.

I would recommend subclassing ScrollPhysics instead for the desired effect. Since scrolling gestures are being overridden, it sounds like there is a lot of custom interaction logic already at play. The Scrollbar checks ScrollPhysics.shouldAcceptUserOffset to determine whether or not it can be dragged. I would look at overriding that method, allowing the drag in the context of scrollbar interactions, and not when whatever custom gesture system is being used is engaged.
